### ISSUE TITLE
In native Wayland mode, allow attaching a file from all xdg-user-dirs (DOCUMENTS, MUSIC, PICTURES, VIDEOS...) and not only DOWNLOAD

### ISSUE THREAD

**Issue by antistress** [flathub/org.mozilla.Thunderbird#246]
Hi,

I know that portals are here for security reasons.

At present time, **Save** and **Attach** seems to have the same access.

From what I understand, **Save** is about writing whereas **attach** is about reading files on the disk.
I guess that security could be adjusted regarding this difference ?

I suggest to let user attach files from most xdg-user-dirs (and not only DOWNLOAD) :  DOCUMENTS, MUSIC, PICTURES, VIDEOS

These bugs may be related (not sure) :
 Can't interact with attachments #19 
 Can't save files / attachments #35 
 Use xdg file chooser (or mention in README) #214 

Thanks

(using TB 91.10.0 flatpak from FlatHub on Debian Testing with GNOME 42.2)

**Comment by Erick555** [comment]
I don't understand the issue. You can save files to any path you want and download files to any path you want through portals. Just tested it and everything works fine if the xdg-portal works on your system. All bugs you mentioned are fixed.

**Comment by antistress** [comment]
![Capture d’écran du 2022-06-15 15-52-15](https://user-images.githubusercontent.com/2436957/173844341-6e0b066d-fb5c-435a-b8bb-ddd82451a0fe.png)

Here's what I get when I try to attach a file
If you say that I should see the other directories, it may a bug related ton my non-default configuration since I moved my xdg-user-dirs onto another disk

nano ~/.config/user-dirs.dirs shows :
XDG_DESKTOP_DIR="$HOME/"
XDG_DOWNLOAD_DIR="/mnt/data/Téléchargements"
XDG_TEMPLATES_DIR="$HOME/Modèles"
XDG_PUBLICSHARE_DIR="$HOME/Public"
XDG_DOCUMENTS_DIR="/mnt/data/Documents"
XDG_MUSIC_DIR="/mnt/data/Musique"
XDG_PICTURES_DIR="/mnt/data/Images"
XDG_VIDEOS_DIR="/mnt/data/Vidéos"

**But** since the change is ok in other apps and **above all** since TB shows my XDG_DOWNLOAD_DIR, I'm not convinced that my non-default configuration interferes

**Comment by antistress** [comment]
And for instance Audacity flatpak shows all my xdg-user-dirs

Thanks

**Comment by antistress** [comment]
Just to be sure, before I follow your link  :
With all those flatpak applications from FlatHub file dialog is fine on my system :
Audacity (wxWidgets 3)
Avidemux (Qt)
Déjà Dup (GTK+3)
FileZilla (wxWidgets 3)
GIMP (GTK+2)
Liferea (GTK+3)
LibreOffice (VCL)
Pitivi (GTK+3)
Transmission (GTK+3)
GNOME Web (GTK+3)

Don't that exculpate the xdg-desktop-portal on my system ?

**Comment by Erick555** [comment]
By fine you mean all your system files are listed on file dialog? Could you check on firefox flatpak as well?

**Comment by antistress** [comment]
Yes, file dialog seems complete with these flatpak applications

Here is what I get with Audacity for an exemple :
![Capture d’écran du 2022-06-16 00-54-44](https://user-images.githubusercontent.com/2436957/173955876-94053dc2-cb97-4c56-9dfc-e6a719229594.png)

It's also ok with flatpak Firefox using the "Save page as" option

Unlike flatpak TB using the Save As > File option

**Comment by Erick555** [comment]
No idea then. It's not reproducible for me - file dialog looks similar to yours from Audacity. Maybe you could try setting `widget.use-xdg-desktop-portal` to true in TB config editor.

**Comment by Erick555** [comment]
Looking closer at your app list I noticed that Audacity has `--filesystem=host` which bypasses portals, LO and gimp don't use portals at all. Not sure about the rest.

**Comment by Erick555** [comment]
Avidemux, dejadup, filezilla, pitivi and transmission also have --filesystem=host so only relevant apps for comparison are liferea, gnome web and ff. Did you ever used override permissions for them? For example what `flatpak info -M org.mozilla.firefox` shows?

**Comment by antistress** [comment]
Ha that's an interesting fact, I had no idea of that setting regarding these software.

Concerning Firefox :

```
$ flatpak info -M org.mozilla.firefox
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;pcsc;cups;
devices=all;
filesystems=xdg-download;
persistent=.mozilla;

[Session Bus Policy]
org.freedesktop.Notifications=talk
org.freedesktop.FileManager1=talk
org.mozilla.firefox.*=own
org.mpris.MediaPlayer2.firefox.*=own
org.mozilla.firefox_beta.*=own
org.freedesktop.ScreenSaver=talk
org.gtk.vfs.*=talk
org.a11y.Bus=talk
org.gnome.SessionManager=talk

[System Bus Policy]
org.freedesktop.NetworkManager=talk

[Environment]
GTK_PATH=/app/lib/gtkmodules
```
![Capture d’écran du 2022-06-16 22-27-22](https://user-images.githubusercontent.com/2436957/174157979-6961b88f-bc35-43a4-8ccb-d3c1f241c729.png)

**Comment by antistress** [comment]
```
$ flatpak info -M org.mozilla.Thunderbird
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;pcsc;cups;
devices=dri;
filesystems=xdg-download;xdg-run/gnupg:ro;~/.gnupg;
persistent=.thunderbird;

[Session Bus Policy]
org.freedesktop.Notifications=talk
org.mozilla.thunderbird.*=own
org.a11y.*=talk

[Environment]
GTK_PATH=/app/lib/gtkmodules
MOZ_ENABLE_WAYLAND=1
```

**Comment by Erick555** [comment]
The file dialog from ff is showing just 2 folders - are those all you have in HOME or is something lost?

**Comment by antistress** [comment]
Yes, these 2 folders are those I have in Home that are not hidden.

I did :
`flatpak --user override --socket=wayland --env=MOZ_ENABLE_WAYLAND=1 org.mozilla.firefox`

And about:support confirms that ff runs in native Wayland mode. FF File dialog is still ok.

Do you want me to run TB in X11 mode with Xwayland ?

**Comment by Erick555** [comment]
Yes, you may try it.

**Comment by antistress** [comment]
Okay, that looks interesting.

I've run $ flatpak --user override --env=MOZ_ENABLE_WAYLAND= org.mozilla.Thunderbird
and checked with xlsclients that TB was running with XWayland
and File dialog is now complete !
![Capture d’écran du 2022-06-18 03-20-30](https://user-images.githubusercontent.com/2436957/174417049-8ed13a56-f07b-4e08-a8af-64e5e47706dc.png)

**Comment by Erick555** [comment]
So it may be a bug in Firefox ESR branch which Thunderbird is based on while it's fixed on Latest FF code. I don't know if I can find relevant change but next major Thunderbird release which will be available in few months should have it fixed.

**Comment by antistress** [comment]
Okay, let's wait and I'l try again then
Thanks !

**Comment by kaimast** [comment]
Did you actually try setting `widget.use-xdg-desktop-portal=true` in the config editor? I still have to do that on every new install of Thunderbird; not sure why though...

**Comment by antistress** [comment]
If you don't mind I will first wait for TB 102 and then, if needed, I'll try your tips (thanks for it)

**Comment by antistress** [comment]
Hi, bug is solved with latest TB flatpak 102.2.0 running in pure Wayland mode :)

**Comment by Erick555** [comment]
That's great to hear! I think we can close it then.
